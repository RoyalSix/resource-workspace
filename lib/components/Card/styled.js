"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Body = exports.Container = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  width: 100%;\n  display: flex;\n  flex-direction: column;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  background-color: white;\n  display: flex;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Container = _styledComponents.default.div(_templateObject());

exports.Container = Container;

var Body = _styledComponents.default.div(_templateObject2());

exports.Body = Body;