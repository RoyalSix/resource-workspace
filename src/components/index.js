export { default as Workspace } from './Workspace';
export { default as MinimizedCardsListUI } from './MinimizedCardsListUI';
