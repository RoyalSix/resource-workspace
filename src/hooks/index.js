export * from './useWindowSize';
export * from './useMinimizedCardsState';
