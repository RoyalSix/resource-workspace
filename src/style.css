.react-grid-item.react-grid-placeholder {
  background-color: grey !important;
}
