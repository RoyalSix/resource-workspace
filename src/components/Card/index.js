import React from 'react';
import { Container, Body } from './styled';

export default function Card({ style, children, ...other }) {
  return (
    <Container style={style} {...other}>
      <Body>
        {children}
      </Body>
    </Container>
  )
}