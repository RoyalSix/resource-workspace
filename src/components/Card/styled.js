import styled from 'styled-components';

export const Container = styled.div`
  background-color: white;
  display: flex;
`;

export const Body = styled.div`
  width: 100%;
  display: flex;
  padding: 10px;
`;

export const Title = styled.div`
  
`;