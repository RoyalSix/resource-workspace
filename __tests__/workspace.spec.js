import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';
import { Workspace } from '../src';
import Card from './utils/Card';

configure({ adapter: new Adapter() });

const absoluteLayout = [
  {
    'w':6, 'h':1, 'x':0, 'y':0, 'i': 'card_0',
  },
  {
    'w':6, 'h':1, 'x':6, 'y':2, 'i': 'card_1',
  },
  {
    'w':6, 'h':1, 'x':0, 'y':2, 'i': 'card_2',
  },
  {
    'w':6, 'h':1, 'x':6, 'y':0, 'i': 'card_3',
  },
  {
    'w':12, 'h':1, 'x':0, 'y':1, 'i': 'card_4',
  },
];

const test_layout = {
  // this line is for relative positioning
  widths: [[1, 1], [1, 1], [1]],
  // uncomment following line to do absolute positioning, for example to restore last saved position
  // absolute: absoluteLayout,
};

describe('testing Workspace', () => {
  it('test widths only', done => {
    const initialLayout = test_layout;
    let found = false;
    generateLayoutTest(initialLayout, 5, found, done);
  });

  it('test widths array and heights array', done => {
    const initialLayout = {
      widths: [[1, 2, 3]],
      heights: [[3, 2, 1]],
    };
    let found = false;
    generateLayoutTest(initialLayout, 3, found, done);
  });

  it('test widths array and numerical height', done => {
    const initialLayout = {
      widths: [[1, 2, 3]],
      heights: 2,
    };
    let found = false;
    generateLayoutTest(initialLayout, 3, found, done);
  });

  it('test absolute layout', done => {
    const match = absoluteLayout;
    const initialLayout = { absolute: absoluteLayout };
    let found = false;
    generateLayoutTest(initialLayout, absoluteLayout.length, found, done, match);
  });
});

//
// helpers
//

function filterLayouts(layout) {
  const include = ['w','h','x','y','i'];
  const filteredLayouts = layout.map(item => {
    const filtered = {};

    for (const key of include) {
      filtered[key] = item[key];
    }
    return filtered;
  });
  return filteredLayouts;
}

function generateLayoutTest(initialLayout, cardCount, found, done, match = null) {
  function onLayoutChange(layout) {
    if (!found) {
      found = true;
      // console.log(`for initial layout: ${JSON.stringify(initialLayout)}`);
      // console.log(`new resource layout: ${JSON.stringify(layout)}`);
      let filtered = filterLayouts(layout);
      // console.log(`filtered: ${JSON.stringify(filtered)}`);

      if (match) {
        expect(filtered).toEqual(match);
      } else {
        expect(filtered).toMatchSnapshot();
      }
      done();
    }
  }

  mount(
    <Workspace gridMargin={[15, 15]} layout={initialLayout} onLayoutChange={onLayoutChange}>
      {getCards(cardCount)}
    </Workspace>,
  );
}

function getCards(cardCount) {
  const cards = [];

  for (let i = 0; i < cardCount; i++) {
    cards.push(
      <Card title={`Card ${i}`} key={`card_${i}`}>`Card ${i} Content`</Card>,
    );
  }

  return cards;
}