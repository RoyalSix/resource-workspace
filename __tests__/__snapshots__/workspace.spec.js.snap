// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Workspace test widths and heights 1`] = `
Array [
  Object {
    "h": 2,
    "i": "1",
    "w": 2,
    "x": 0,
    "y": 0,
  },
  Object {
    "h": 2,
    "i": "2",
    "w": 4,
    "x": 2,
    "y": 0,
  },
  Object {
    "h": 2,
    "i": "3",
    "w": 6,
    "x": 6,
    "y": 0,
  },
]
`;

exports[`testing Workspace test widths only 1`] = `
Array [
  Object {
    "h": 1,
    "i": "1",
    "w": 6,
    "x": 0,
    "y": 0,
  },
  Object {
    "h": 1,
    "i": "2",
    "w": 6,
    "x": 6,
    "y": 0,
  },
  Object {
    "h": 1,
    "i": "3",
    "w": 6,
    "x": 0,
    "y": 1,
  },
  Object {
    "h": 1,
    "i": "4",
    "w": 6,
    "x": 6,
    "y": 1,
  },
  Object {
    "h": 1,
    "i": "5",
    "w": 12,
    "x": 0,
    "y": 2,
  },
]
`;
